### ISSUE TITLE
MU4 crash in Linux

### ISSUE THREAD

**Issue by krtschil** [musescore/MuseScore#13698]
**Description:**
Today I experienced a crash running the Nightly build from Oct, 10th.
Upon closing the open file it crashed (after refusing to save changes to the open file) with the following message:

/tmp/.mount_MuseSc7ltvD2/AppRun: Zeile 26: 44217 Speicherzugriffsfehler "${APPDIR}/bin/mscore-portable-nightly" "$@"

OS: Ubuntu 20.04
Appimage
=== Started MuseScore 4.0.0-Development, build number 3216906122 ===

Logfile from that session attached.

**To Reproduce**
1. Open first score (Pachelbel)
2. Play it by clicking the play button
3. While playing open the mixer (F10)
4. Press CTRL-S to save the score
5. Close the score (click the X)
6. Open second score (Schumann from 3.6.0)
7. Play it by clicking the play button
8. Close the score by clicking the X and click "Don't save"
9. Crash (app disappeared completely)

I could reproduce the crash once but no luck afterwards so I am not sure whether the steps above lead to the crash. The only sure thing is step 9 (closing the changed score from Schumann).

[MS4.zip](https://github.com/musescore/MuseScore/files/9756126/MS4.zip)

Klaus

**Comment by lyrra** [comment]
Can reproduce sometimes on linux.
<summary>backtrace<details>

```
try-1:
#0  0x0000555558defea0 in typeinfo for std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2> ()
#1  0x0000555555fae209 in mu::appshell::NavigableAppMenuModel::restoreMUNavigationSystemState (this=0x55555a4bf8e0) at /d/shrek/ms/ms4/src/appshell/view/navigableappmenumodel.cpp:469
#2  0x0000555555facc28 in mu::appshell::NavigableAppMenuModel::setOpenedMenuId (this=0x55555a4bf8e0, openedMenuId=...) at /d/shrek/ms/ms4/src/appshell/view/navigableappmenumodel.cpp:147
#3  0x0000555555f96e00 in mu::appshell::NavigableAppMenuModel::qt_static_metacall (_o=0x55555a4bf8e0, _c=QMetaObject::WriteProperty, _id=2, _a=0x7fffffffaf90)
    at /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/appshell/appshell_autogen/OTUN7GXZT4/moc_navigableappmenumodel.cpp:230
#4  0x0000555555f9705d in mu::appshell::NavigableAppMenuModel::qt_metacall (this=0x55555a4bf8e0, _c=QMetaObject::WriteProperty, _id=2, _a=0x7fffffffaf90)
    at /d/shrek/ms/ms4/builds/Linux-Qt-usr-Make-Debug/src/appshell/appshell_autogen/OTUN7GXZT4/moc_navigableappmenumodel.cpp:281
#5  0x00007ffff75b5e02 in QV4::QObjectWrapper::setProperty(QV4::ExecutionEngine*, QObject*, QQmlPropertyData*, QV4::Value const&) () from /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5

try-2:
#0  0x0000555558defea0 in typeinfo for std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2> ()
#1  0x0000555555fae209 in mu::appshell::NavigableAppMenuModel::restoreMUNavigationSystemState (this=0x55555a4bf8d0)
    at /d/shrek/ms/ms4/src/appshell/view/navigableappmenumodel.cpp:469
#2  0x0000555555fad563 in mu::appshell::NavigableAppMenuModel::processEventForAppMenu (this=0x55555a4bf8d0, event=0x7fffffffd720)
    at /d/shrek/ms/ms4/src/appshell/view/navigableappmenumodel.cpp:305
#3  0x0000555555facfe5 in mu::appshell::NavigableAppMenuModel::eventFilter (this=0x55555a4bf8d0, watched=0x55555942af30, event=0x7fffffffd720)
    at /d/shrek/ms/ms4/src/appshell/view/navigableappmenumodel.cpp:197
#4  0x00007ffff5e0cc06 in QCoreApplicationPrivate::sendThroughApplicationEventFilters(QObject*, QEvent*) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5

```

</details></summary>

**Comment by Tantacrul** [comment]
@DmitryArefiev - can you take a peek and potentially assign (when you have time)?

**Comment by DmitryArefiev** [comment]
@Tantacrul Yes, I was able to reproduce it on **Linux**!

**Comment by krtschil** [comment]
Please see my comment in #13666, maybe that crash is actually related to this crash

**Comment by Eism** [comment]
@krtschil 
Please check the build from https://github.com/musescore/MuseScore/pull/13774

**Comment by krtschil** [comment]
@Eism Sorry, I do not see where to download the build from you mention above

**Comment by cbjeukendrup** [comment]
@krtschil Here's a direct link to the download page for Linux for that PR: https://github.com/musescore/MuseScore/actions/runs/3255818196
At the bottom of that page, in the "Artifacts" section, you can download the build.

We should probably create some documentation for how to find such a build yourself; I'll do that later.

**Comment by HemantAntony** [comment]
@cbjeukendrup A coincidence. I was working on the wiki. Here is the Wiki page: https://github.com/musescore/MuseScore/wiki/Pull-request-testing

**Comment by krtschil** [comment]
Test results with the build above:
1. Opened score from Pachelbel, played it, pressed F10 to open Mixer, CTRL-S to save, pressed ESC to stop playing, closed the score (hitting the X)
2. Opened score from Schumann (MU4 version), played it, pressed ESC to stop and closed the score (hitting the X)
3. Opened score from Schuman using the file open dialog  and selected the MU3 version
4. Played it, stopped with ESC and hit X to close the score (denying to save changes), --> CRASH

This is not always reproducable. I saw the crash several times but not every time. 
[MSD4-1.zip](https://github.com/musescore/MuseScore/files/9792642/MSD4-1.zip)

**Comment by krtschil** [comment]
Forgot to upload the logfiles:

[logfiles.zip](https://github.com/musescore/MuseScore/files/9794433/logfiles.zip)

**Comment by Eism** [comment]
@krtschil Please check this build https://github.com/musescore/MuseScore/suites/8809889681/artifacts/400861796

**Comment by DmitryArefiev** [comment]
Fixed in #13774
